run_bot_streaming.sh

```
@@ -0,0 +1,283 @@
+    local main_port=$(find_available_port 3000)
+    if [ $? -ne 0 ]; then
+        print_error "Failed to allocate port for bot instance"
+        exit 1
+    fi
+
+    print_info "📡 Bot API will be accessible on port $main_port"
+
+    # Setup Docker args
+    local docker_args="-p $main_port:3000"
+    if [ "$debug_mode" = "true" ]; then
+        local vnc_port=$(find_available_port 5900)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)